system/service/apigw_filter.go

```diff
@@ -268,10 +260,6 @@ func (svc *apigwFilter) Search(ctx context.Context, filter types.ApigwFilterFilt
 
 	// For each fetched item, store backend will check if it is valid or not
 	filter.Check = func(res *types.ApigwFilter) (bool, error) {
-		if !svc.ac.CanReadApigwFilter(ctx, res) {
-			return false, nil
-		}
-
```

Comment:
If the rest of the operations check access based on the API gateway route, this should also do that.
That said, it might be a bit annoying as we need to fetch the route as well.

You could preload all of the routes, throw them in a hashmap and then use that here.
Alternatively, we could also limit route filtering to allow searching only over a specific route which would make this trivial to handle.

The same applies to the following two deleted blocks.